### PR TITLE
feat: add bridge for entity audit

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |  none
| License       | MIT
| Doc PR        | todo

add bridge to https://github.com/simplethings/EntityAudit
- [x] add tests

### CODE REVIEW

**Comment by Simperfit** [comment]
This is RFR.

**Comment by dunglas** [comment]
What do you think about adding support this in a external lib + bundle? IMO it's a too specific library to add this bridge in core. An alternative would be to add this bridge directly to `simplethings/EntityAudit`.

File: src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass) : ResourceMetadata
+    {
+        $resourceClassMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+        if (true === $this->auditManager->getMetadataFactory()->isAudited($resourceClass)) {
+            $resourceAuditedMetadata = new ResourceMetadata(
+                    $resourceClassMetadata->getShortName(),
```
Comment (sroze) [inline src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php]: Not convinced about the CS here :)

File: src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * Creates a resource metadata from EntityAudit Entity.
+ *
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+final class EntityAuditResourceMetadataFactory implements ResourceMetadataFactoryInterface
+{
+    private $resourceMetadataFactory;
+    private $auditManager;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory = null, AuditManager $auditManager)
```
Comment (sroze) [inline src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php]: The first argument can't be `null` here

File: src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php (hunk 1 above)
Comment (Simperfit) [inline src/Bridge/SimpleThingsEntityAudit/Metadata/Resource/Factory/EntityAuditResourceMetadataFactory.php]: I feel like it is more readable like this, and there is no problem with php-cs / styleci, what is the problem for you in here ? 🐰

**Comment by Simperfit** [comment]
will do that

**Pull request closed without merging**